targettests/execution/custom_operation_adj.cpp

```diff
@@ -7,6 +7,10 @@
  ******************************************************************************/
 
 // RUN: nvq++ %cpp_std --enable-mlir %s -o %t && %t | FileCheck %s
+// RUN: nvq++ %cpp_std --target ionq --emulate %s -o %t && %t | FileCheck %s
+// RUN: nvq++ %cpp_std --target iqm --emulate --iqm-machine Apollo %s -o %t && %t | FileCheck %s
+// RUN: nvq++ %cpp_std --target oqc --emulate %s -o %t && %t | FileCheck %s
+// RUN: nvq++ %cpp_std --target quantinuum --emulate %s -o %t && %t | FileCheck %s
```

Comment:
It seems odd that none of these change the output in any way or if they do that we wouldn't check that the changes are correct and what is expected.
Corollary: if these new command lines don't exercise any new functionality, there is no need to add them to the regression tests.

Reply:
These should haven failing test(s) prior to the changes in this PR. There was a failing target test in Python but missing in C++. The test here is that the result is same across simulation targets which directly apply the custom matrix and hardware emulation targets which now decompose that custom matrix.

Reply:
I forgot that these tests are run in both a normal and a weird way.